mobile/src/splash-screen/splash-screen.tsx

```diff
@@ -1,16 +1,92 @@
 import React from 'react';
-import { View, Image } from 'react-native';
+import { Image, Platform, View } from 'react-native';
 import { splashScreenStyles } from './styles';
+import { FingerprintPopupAndroid } from '../fingerprint-popup/fingerprint-popup.android';
+import { Action } from 'redux';
+import { startLoginProcess, startLogoutProcess } from '../reducers/auth/auth.action';
+import { connect, Dispatch } from 'react-redux';
 
-export class SplashScreen extends React.Component<{}> {
+//============================================================================
+interface SplashScreenState {
+    fingerprintPopupVisible: boolean;
+}
+
+//============================================================================
+interface SplashScreenDispatchProps {
+    login: () => void;
+    logout: () => void;
+}
+
+//============================================================================
+class SplashScreenImpl extends React.Component<SplashScreenDispatchProps, SplashScreenState> {
+
+    //----------------------------------------------------------------------------
+    constructor(props: SplashScreenDispatchProps) {
+        super(props);
 
+        this.onPopupClosed = this.onPopupClosed.bind(this);
```

Comment:
Hmm, why is it so? I’m okay with lambdas, but why aren’t methods?

Reply:
When you use simple methods or lambdas it'll compile to
```
function Foo {
//either
this.lambda = () => {}; //autocapture this, create a function instance on each class instance
}
//or
Foo.prototype.method = function() {} // put shared function into prototype and call it every time
```
If you do like you did, it'll have both instance function defined AND prototype method, so in runtime it'll possible to get both these functions and they work differently. You should not notice any problems in your case, but when it comes to inheritance or some runtime magic, it can bit you.
That's from JS point of view. From TS, `this` typing (strict when lambda, not-so-strict when a member) also exist and you sort of deceive them.

Reply:
Addressed in 29b70b8f7989b0d930725c2c79041cf00c4c8d12